org.opentosca.bus.management.api.osgievent/src/org/opentosca/bus/management/api/osgievent/MBEventHandler.java

```diff
@@ -48,6 +90,54 @@
 
     private EventAdmin eventAdmin;
 
+    private ConsumerTemplate invokePlan(String operationName, String messageID, boolean async, String serviceInstanceID,
+                                        Object message, CSARID csarID, QName planID, String planLanguage) {
+        MBEventHandler.LOG.debug("Plan invocation is asynchronous: {}", async);
+
+        // create the headers for the Exchange which is send to the Management Bus
+        final Map<String, Object> headers = new HashMap<>();
+        headers.put(MBHeader.CSARID.toString(), csarID);
+        headers.put(MBHeader.PLANID_QNAME.toString(), planID);
+        headers.put(MBHeader.OPERATIONNAME_STRING.toString(), operationName);
+        headers.put(MBHeader.PLANCORRELATIONID_STRING.toString(), messageID);
+        headers.put("OPERATION", OsgiEventOperations.INVOKE_PLAN.getHeaderValue());
+        headers.put("PlanLanguage", planLanguage);
+
+
+        if (message instanceof HashMap) {
+            MBEventHandler.LOG.debug("Invocation body is of type HashMap.");
+
+            if (serviceInstanceID != null) {
+                URI serviceInstanceURI;
+                try {
+                    serviceInstanceURI = new URI(serviceInstanceID);
+                    headers.put(MBHeader.SERVICEINSTANCEID_URI.toString(), serviceInstanceURI);
+                }
+                catch (final URISyntaxException e) {
+                    MBEventHandler.LOG.warn("Could not generate service instance URL: {}", e.getMessage(), e);
+                }
+            } else {
+                MBEventHandler.LOG.warn("Service instance ID is null.");
+            }
+        } else {
+            MBEventHandler.LOG.warn("Invocation body is of type: {}", message.getClass());
+        }
+
+        // templates to communicate with the Management Bus
+        final ProducerTemplate template = Activator.camelContext.createProducerTemplate();
+        final ConsumerTemplate consumer = Activator.camelContext.createConsumerTemplate();
```

Comment:
this is never used before being returned. AFAICT this could even be stored in an instance variable and doesn't need to be grabbed for every method call.

The javadoc for both `createProducerTemplate()` and `createConsumerTemplate()` notes that one should call `Service.stop()` when you're done with the template, so it seems that this could be a memory leak...

Reply:
It is never used because other method use the returned consumer template and stop it (e.g. Line 193)